src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java

```diff
@@ -282,11 +282,11 @@ public CpsBodyInvoker newBodyInvoker() {
         if (body == null) {
             throw new IllegalStateException("There is no body to invoke");
         }
-        return newBodyInvoker(body);
+        return newBodyInvoker(body, false);
```

Comment:
Note that this is the normal case—everything _except_ `parallel` and `load`. Just normal block-scoped steps like `node`, (new) `stage`, etc. They already unexported their closure when the step completes in `CpsStepContext.scheduleNextRun` (whether or not the body was ever invoked, or invoked multiply).